pkg/newlog/cmd/newlogd.go

```diff
@@ -763,6 +763,10 @@ func writelogFile(logChan <-chan inputEntry, moveChan chan fileChanInfo) {
 				appM = getAppStatsMap(appuuid)
 			}
 			timeS := getPtypeTimestamp(entry.timestamp)
+			if timeS == nil {
+				// Can't parse timestamp, silently drop the entry
```

Comment:
Not sure how that can help. Especially having that we *are* in attempt to write a log message.

Reply:
interesting point @rouming, but at this stage we are trying to write to the file, right? a log memory buffer should still be available, so we could use log.Fatal(), etc; another thing is that this failing will be due to the format (lack of timestamp) rather then another more critical problem. That said, I think is worth to log a warning attempt to write something invalid to the logs...  it should work, or am I missing something?

Reply:
I can't tell whether we need to support an empty timestamp but I think the validation should actually happen earlier where we currently check
		if logInfo.Time == "" && strings.HasSuffix(msgTime, "Z") {
			logInfo.Time = msgTime
		}
The other places where the code produces an inputEntry it is correctly formated.

To be this would be easiest if we change the timestamp field in inputEntry to be of type time.Time and then have the producers in this file fill that in. And we should either allow zero (means Jan 1 1970) or if zero then set it to time.Now().